e2e/tests/search.e2e.spec.js

```diff
@@ -0,0 +1,127 @@
+
+        // Wait for search to complete
+        await waitForSearchCompletion(page);
+
+        // Get the search results
+        const searchResults = await page.locator(searchResultSelector);
+
+        // Verify that no results are found
+        expect(await searchResults.count()).toBe(0);
+        expect(await page.locator('text=No results found').count()).toBe(1);
+    });
+
+    test('Validate single object in search result [full search]', async ({ page }) => {
+        //Go to baseURL
+        await page.goto("/", { waitUntil: "networkidle" });
+
+        // Create a folder object
+        const folderName = 'testFolder';
+        await createFolderObject(page, folderName);
+
+        // Full search for object
+        await page.type("input[type=search]", folderName);
+
+        // Wait for search to complete
+        await waitForSearchCompletion(page);
+
+        // Get the search results
+        const searchResults = await page.locator(searchResultSelector);
+
+        // Verify that one result is found
+        expect(await searchResults.count()).toBe(1);
+        await expect(searchResults).toHaveText(folderName);
+    });
+
+    test("Validate multiple objects in search results [partial search]", async ({ page }) => {
+        test.info().annotations.push({
+            type: 'issue',
+            description: 'https://github.com/nasa/openmct/issues/4667'
+        });
+
+        // Go to baseURL
+        await page.goto("/", { waitUntil: "networkidle" });
+
+        // Create folder objects
+        const folderName = "e928a26e-e924-4ea0";
+        const folderName2 = "e928a26e-e924-4001";
+
+        await createFolderObject(page, folderName);
+        await createFolderObject(page, folderName2);
+
+        // Partial search for objects
+        await page.type("input[type=search]", 'e928a26e');
+
+        // Wait for search to finish
+        await waitForSearchCompletion(page);
+
+        // Get the search results
+        const searchResults = await page.locator(searchResultSelector);
+
+        // Verify that the search result/s correctly match the search query
+        expect(await searchResults.count()).toBe(2);
+        await expect(await searchResults.first()).toHaveText(folderName);
+        await expect(await searchResults.last()).toHaveText(folderName2);
+    });
+});
+
+async function createFolderObject(page, folderName) {
+    // Open Create menu
+    await page.locator('button:has-text("Create")').click();
+
+    // Select Folder object
+    await page.locator('text=Folder').nth(1).click();
+
+    // Click folder title to enter edit mode
+    await page.locator('text=Properties Title Notes >> input[type="text"]').click();
+
+    // Enter folder name
+    await page.locator('text=Properties Title Notes >> input[type="text"]').fill(folderName);
+
+    // Create folder object
+    await page.locator('text=OK').click();
+}
+
+async function waitForSearchCompletion(page) {
```

Comment:
This is great. Another alternative is to .waitforselector

Reply:
Thanks for the feedback. Updated accordingly.